addon/addon.cpp

```diff
@@ -0,0 +1,80 @@
+#include <cstdio>
+#include <napi.h>
+
+#include <vencord/audio.hpp>
+
+#include <range/v3/view.hpp>
+#include <range/v3/algorithm.hpp>
+
+Napi::Array venmic_list(const Napi::CallbackInfo &info)
+{
+    auto env = info.Env();
+
+    auto list = vencord::audio::get().list();
+    auto rtn  = Napi::Array::New(env, list.size());
+
+    auto convert = [&](const auto &item)
+    {
+        auto obj = Napi::Object::New(env);
+
+        obj.Set("name", Napi::String::New(env, item.name));
+        obj.Set("id", Napi::Number::New(env, item.id));
+
+        return obj;
+    };
+
+    auto add = [&](const auto &item)
+    {
+        auto [index, value] = item;
+        rtn.Set(index, value);
+    };
+
+    ranges::for_each(list                                    //
+                         | ranges::views::transform(convert) //
+                         | ranges::views::enumerate,
+                     add);
+
+    return rtn;
+}
+
+Napi::Boolean venmic_link(const Napi::CallbackInfo &info)
+{
+    auto env = info.Env();
+
+    if (info.Length() != 1)
+    {
+        Napi::Error::New(env, "[venmic] 'link' requires one argument");
```

Comment:
shouldn't this be
```suggestion
        Napi::Error::New(env, "[venmic] 'link' requires one argument").ThrowAsJavaScriptException();
```

Reply:
Yep, good catch, copied the usage from the cmake-js example which seems to be wrong